pkg/orchestrator/util.go

```diff
@@ -112,3 +112,16 @@ func getTelemetryTags(n NodeType) []string {
 		strings.ToLower(n.String()),
 	}
 }
+
+// ChunkRange returns the chunk start and end for an iteration.
+func ChunkRange(chunkLen, chunkCount, chunkSize, counter int) (int, int) {
```

Comment:
Could we add unit tests for this?

Reply:
Good point! Let me add one 👌

Reply:
Added one in https://github.com/DataDog/datadog-agent/pull/8185/commits/d2069c51dc0201af3385e3e468f564434bffd56f . Added a note over the function regarding the chunkCount input.

Additionally I extraced `GroupSize` into a dedicated function with unit tests: d2b36802e2b56273143450c2e9088becf96d8cfc